src/common/render_tree.cpp

```diff
@@ -204,6 +204,26 @@ unique_ptr<RenderTree> RenderTree::CreateRenderTree(const ProfilingNode &op) {
 	return CreateTree<ProfilingNode>(op);
 }
 
+void RenderTree::SanitizeKeyNames() {
```

Comment:
The QueryProfiler handles the JSON version of the EXPLAIN ANALYZE output, which also needs this I think

```sql
statement ok
PRAGMA enable_verification

statement ok
CREATE TABLE integers AS SELECT * FROM range(100) tbl(i);

statement error
EXPLAIN (FORMAT JSON, ANALYZE) SELECT SUM(i) FROM (SELECT * FROM integers i1, integers i2 UNION ALL SELECT * FROM integers i1, integers i2);
----
```

```
explain_key     explain_value
VARCHAR VARCHAR
[ Rows: 1]
analyzed_plan   {
    "query": "",
    "cpu_time": 0.0,
    "extra_info": {},
    "cumulative_cardinality": 0,
    "operator_cardinality": 0,
    "cumulative_rows_scanned": 0,
    "operator_rows_scanned": 0,
    "operator_timing": 0.0,
    "timings": [
        {
            "annotation": "optimizer",
            "timing": 0.001717
        },
        {
            "annotation": "optimizer > build_side_probe_side",
            "timing": 0.000072
        },
        {
            "annotation": "optimizer > column_lifetime",
            "timing": 0.000019999999999999998
        },
        {
            "annotation": "optimizer > common_aggregate",
            "timing": 0.000012
        },
        {
            "annotation": "optimizer > common_subexpressions",
            "timing": 0.000014
        },
        {
            "annotation": "optimizer > cte_filter_pusher",
            "timing": 0.000002
        },
        {
            "annotation": "optimizer > deliminator",
            "timing": 0.000008
        },
        {
            "annotation": "optimizer > duplicate_groups",
            "timing": 0.000013
        },
        {
            "annotation": "optimizer > expression_rewriter",
            "timing": 0.00003
        },
        {
            "annotation": "optimizer > filter_pullup",
            "timing": 0.000003
        },
        {
            "annotation": "optimizer > filter_pushdown",
            "timing": 0.000047
        },
        {
            "annotation": "optimizer > in_clause",
            "timing": 0.000008
        },
        {
            "annotation": "optimizer > join_filter_pushdown",
            "timing": 0.000004
        },
        {
            "annotation": "optimizer > join_order",
            "timing": 0.000374
        },
        {
            "annotation": "optimizer > limit_pushdown",
            "timing": 0.000002
        },
        {
            "annotation": "optimizer > regex_range",
            "timing": 0.000002
        },
        {
            "annotation": "optimizer > reorder_filter",
            "timing": 0.000013
        },
        {
            "annotation": "optimizer > statistics_propagation",
            "timing": 0.000085
        },
        {
            "annotation": "optimizer > top_n",
            "timing": 0.000002
        },
        {
            "annotation": "optimizer > unnest_rewriter",
            "timing": 0.000003
        },
        {
            "annotation": "optimizer > unused_columns",
            "timing": 0.000041
        },
        {
            "annotation": "physical_planner",
            "timing": 0.000232
        },
        {
            "annotation": "physical_planner > column_binding",
            "timing": 0.000015
        },
        {
            "annotation": "physical_planner > create_plan",
            "timing": 0.00017
        },
        {
            "annotation": "physical_planner > resolve_types",
            "timing": 0.00001
        },
        {
            "annotation": "planner",
            "timing": 0.0007549999999999999
        },
        {
            "annotation": "planner > binder",
            "timing": 0.000744
        }
    ],
    "children": [
        {
            "name": "EXPLAIN_ANALYZE",
            "operator_timing": 0.000001,
            "operator_rows_scanned": 0,
            "cumulative_rows_scanned": 0,
            "operator_cardinality": 0,
            "cumulative_cardinality": 0,
            "extra_info": {},
            "cpu_time": 0.0,
            "children": [
                {
                    "name": "UNGROUPED_AGGREGATE",
                    "cpu_time": 0.0,
                    "extra_info": {
                        "Aggregates": "sum_no_overflow(#0)"
                    },
                    "cumulative_cardinality": 0,
                    "operator_cardinality": 1,
                    "cumulative_rows_scanned": 0,
                    "operator_rows_scanned": 0,
                    "operator_timing": 0.0003589999999999995,
                    "children": [
                        {
                            "name": "PROJECTION",
                            "operator_timing": 0.004364000000000003,
                            "operator_rows_scanned": 0,
                            "cumulative_rows_scanned": 0,
                            "operator_cardinality": 20000,
                            "cumulative_cardinality": 0,
                            "extra_info": {
                                "__projections__": "i",
                                "__estimated_cardinality__": "20000"
                            },
                            "cpu_time": 0.0,
                            "children": [
                                {
                                    "name": "UNION",
                                    "cpu_time": 0.0,
                                    "extra_info": {},
                                    "cumulative_cardinality": 0,
                                    "operator_cardinality": 0,
                                    "cumulative_rows_scanned": 0,
                                    "operator_rows_scanned": 0,
                                    "operator_timing": 0.0,
                                    "children": [
                                        {
                                            "name": "PROJECTION",
                                            "operator_timing": 0.002186000000000003,
                                            "operator_rows_scanned": 0,
                                            "cumulative_rows_scanned": 0,
                                            "operator_cardinality": 10000,
                                            "cumulative_cardinality": 0,
                                            "extra_info": {
                                                "__projections__": "i",
                                                "__estimated_cardinality__": "10000"
                                            },
                                            "cpu_time": 0.0,
                                            "children": [
                                                {
                                                    "name": "CROSS_PRODUCT",
                                                    "cpu_time": 0.0,
                                                    "extra_info": {},
                                                    "cumulative_cardinality": 0,
                                                    "operator_cardinality": 10000,
                                                    "cumulative_rows_scanned": 0,
                                                    "operator_rows_scanned": 0,
                                                    "operator_timing": 0.00022499999999999964,
                                                    "children": [
                                                        {
                                                            "name": "SEQ_SCAN ",
                                                            "operator_timing": 0.0000049999999999999996,
                                                            "operator_rows_scanned": 100,
                                                            "cumulative_rows_scanned": 0,
                                                            "operator_cardinality": 100,
                                                            "cumulative_cardinality": 0,
                                                            "extra_info": {
                                                                "__text__": "integers",
                                                                "Projections": "",
                                                                "__estimated_cardinality__": "100"
                                                            },
                                                            "cpu_time": 0.0,
                                                            "children": []
                                                        },
                                                        {
                                                            "name": "SEQ_SCAN ",
                                                            "operator_timing": 0.000051,
                                                            "operator_rows_scanned": 100,
                                                            "cumulative_rows_scanned": 0,
                                                            "operator_cardinality": 100,
                                                            "cumulative_cardinality": 0,
                                                            "extra_info": {
                                                                "__text__": "integers",
                                                                "Projections": "i",
                                                                "__estimated_cardinality__": "100"
                                                            },
                                                            "cpu_time": 0.0,
                                                            "children": []
                                                        }
                                                    ]
                                                }
                                            ]
                                        },
                                        {
                                            "name": "PROJECTION",
                                            "operator_timing": 0.002168000000000002,
                                            "operator_rows_scanned": 0,
                                            "cumulative_rows_scanned": 0,
                                            "operator_cardinality": 10000,
                                            "cumulative_cardinality": 0,
                                            "extra_info": {
                                                "__projections__": "i",
                                                "__estimated_cardinality__": "10000"
                                            },
                                            "cpu_time": 0.0,
                                            "children": [
                                                {
                                                    "name": "CROSS_PRODUCT",
                                                    "cpu_time": 0.0,
                                                    "extra_info": {},
                                                    "cumulative_cardinality": 0,
                                                    "operator_cardinality": 10000,
                                                    "cumulative_rows_scanned": 0,
                                                    "operator_rows_scanned": 0,
                                                    "operator_timing": 0.0002469999999999996,
                                                    "children": [
                                                        {
                                                            "name": "SEQ_SCAN ",
                                                            "operator_timing": 0.000014000000000000001,
                                                            "operator_rows_scanned": 100,
                                                            "cumulative_rows_scanned": 0,
                                                            "operator_cardinality": 100,
                                                            "cumulative_cardinality": 0,
                                                            "extra_info": {
                                                                "__text__": "integers",
                                                                "Projections": "",
                                                                "__estimated_cardinality__": "100"
                                                            },
                                                            "cpu_time": 0.0,
                                                            "children": []
                                                        },
                                                        {
                                                            "name": "SEQ_SCAN ",
                                                            "operator_timing": 0.00005,
                                                            "operator_rows_scanned": 100,
                                                            "cumulative_rows_scanned": 0,
                                                            "operator_cardinality": 100,
                                                            "cumulative_cardinality": 0,
                                                            "extra_info": {
                                                                "__text__": "integers",
                                                                "Projections": "i",
                                                                "__estimated_cardinality__": "100"
                                                            },
                                                            "cpu_time": 0.0,
                                                            "children": []
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    ]
}
```

Reply:
Good point - I've done the sanitization there as well